src/routes/news/reducers/index.js

```diff
@@ -0,0 +1,54 @@
+// @flow
+
+import { Payload, TunewsModel, TUNEWS_ENDPOINT_NAME } from '@integreat-app/integreat-api-client'
+import type { PayloadDataType } from '../../../modules/app/PayloadDataType'
+import { START_FETCH_TUNEWS, FINISH_FETCH_TUNEWS } from '../actions/fetchMoreTunews'
+import { startFetchMoreActionName } from '../actions/startFetchMoreAction'
+import { finishFetchMoreActionName } from '../actions/finishFetchMoreAction'
+
+type TuNewsFetchActionType<T: PayloadDataType> = { type: string, payload: Payload<T> }
+
+const defaultState = new Payload(false, false, [])
+
+const fetchTunewsReducer = (
+  state: Payload<Array<TunewsModel>> = defaultState,
+  action: TuNewsFetchActionType<TunewsModel>
+) => {
+  switch (action.type) {
+    case START_FETCH_TUNEWS:
+      return {
+        data: [],
+        requestUrl: action.payload.requestUrl,
+        isFetching: state.isFetching,
+        isFetchingFirstTime: true
+      }
+    case FINISH_FETCH_TUNEWS:
+      return {
+        data: [...action.payload.data],
+        requestUrl: action.payload.requestUrl,
+        isFetching: state.isFetching,
+        hasMore: action.payload.data.length !== 0,
```

Comment:
Is this correct? Seems weird to me.

Reply:
Yeah. If the last request(while scrolling down) is an empty array then `hasMore` is set to `false` to stop sending more requests